src/sqlancer/ProviderAdapter.java

```diff
@@ -67,7 +91,47 @@ public Reproducer<G> generateAndTestDatabase(G globalState) throws Exception {
         return null;
     }
 
-    protected abstract void checkViewsAreValid(G globalState);
+    @Override
+    public void qpg(G globalState) throws Exception {
+        try {
+            generateDatabase(globalState);
+            checkViewsAreValid(globalState);
+            globalState.getManager().incrementCreateDatabase();
```

Comment:
The logic here seems to be the same as for `generateAndTestDatabase`. Perhaps we can have one common method `generateDatabase` and then the two methods, for testing with QPG and one with randomly generating tests?